### ISSUE TITLE
Migrations :: non-empty migration created when trying to add new migration right after creating ASPNET 5 Starter Web template

### ISSUE THREAD

**Issue by maumar** [dotnet/efcore#1982]
Repro:
- create ASP NET 5 Starter Web project
- create new migration

Generates:

```
        public override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AlterColumn("AspNetRoleClaims", "Id", x => x.Column("int", annotations: new Dictionary<string, string> { { "SqlServer:ValueGeneration", "Identity" } }));
            migrationBuilder.AlterColumn("AspNetUserClaims", "Id", x => x.Column("int", annotations: new Dictionary<string, string> { { "SqlServer:ValueGeneration", "Identity" } }));
        }
```

When trying to apply this, the following exception is thrown:

System.Data.SqlClient.SqlException (0x80131904): Incorrect syntax near the keyword 'IDENTITY'.
At line:1 char:1
- dnx . ef migration apply
- ~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : NotSpecified: (System.Data.Sql...ord 'IDENTITY'.:String) [], RemoteException
  - FullyQualifiedErrorId : NativeCommandError

Incorrect syntax near the keyword 'IDENTITY'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapC
loseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Microsoft.Data.Entity.Relational.SqlStatementExecutor.<>c__DisplayClass7_0.<ExecuteNonQuery>b__0()
   at Microsoft.Data.Entity.Relational.SqlStatementExecutor.Execute(IRelationalConnection connection, Func`1 action)
   at Microsoft.Data.Entity.Relational.SqlStatementExecutor.ExecuteNonQuery(IRelationalConnection connection, DbTransaction transaction, IEnumerable`1 sqlBatches)
   at Microsoft.Data.Entity.Relational.Migrations.Migrator.Execute(IEnumerable`1 sqlBatches, Boolean ensureDatabase)
   at Microsoft.Data.Entity.Relational.Migrations.Migrator.ApplyMigrations(String targetMigration)
   at Microsoft.Data.Entity.Commands.MigrationTool.ApplyMigration(String migrationName, String contextTypeName)
   at Microsoft.Data.Entity.Commands.Program.<>c__DisplayClass10_0.<ApplyMigration>b__0()
   at Microsoft.Data.Entity.Commands.Program.ExecuteInDirectory(String startupProject, Func`1 invoke)
   at Microsoft.Data.Entity.Commands.Program.ApplyMigration(String migration, String context, String startupProject)
   at Microsoft.Data.Entity.Commands.Program.<>c__DisplayClass7_2.<Main>b__8()
   at Microsoft.Framework.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.Data.Entity.Commands.Program.Main(String[] args)
ClientConnectionId:0fd27495-2e0a-40b9-9f5a-b2b02c4d6b90

**Comment by divega** [comment]
@maumar I talked to @bricelam and this apparently was fixed in more recent bits. Could you double-check with the latest?

**Comment by maumar** [comment]
Verified that its fixed in new bits, closing

**Comment by fradique** [comment]
Something similar happening on beta6
after opening the newly created project from web site template, a non-empty migration is generated and cannot be applied:

System.Data.SqlClient.SqlException (0x80131904): Incorrect syntax near the keywo
rd 'IDENTITY'.
Incorrect syntax near the keyword 'IDENTITY'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolea
n breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception
, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObj
ect stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand
 cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler,
TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName,
Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSou
rce`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean
asyncWrite)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Microsoft.Data.Entity.Infrastructure.SqlStatementExecutor.<>c__DisplayClas
s7_0.<ExecuteNonQuery>b__0()
   at Microsoft.Data.Entity.Infrastructure.SqlStatementExecutor.Execute(IRelatio
nalConnection connection, Func`1 action)
   at Microsoft.Data.Entity.Infrastructure.SqlStatementExecutor.ExecuteNonQuery(
IRelationalConnection connection, DbTransaction transaction, IEnumerable`1 sqlBa
tches)
   at Microsoft.Data.Entity.Migrations.Migrator.Execute(IEnumerable`1 sqlBatches
, Boolean ensureDatabase)
   at Microsoft.Data.Entity.Migrations.Migrator.ApplyMigrations(String targetMig
ration)
   at Microsoft.Data.Entity.Commands.MigrationTool.ApplyMigration(String migrati
onName, String contextTypeName, String startupAssemblyName)
   at Microsoft.Data.Entity.Commands.Program.<>c__DisplayClass13_0.<ApplyMigrati
on>b__0()
   at Microsoft.Data.Entity.Commands.Program.Execute(String startupProject, Func
`1 invoke)
   at Microsoft.Data.Entity.Commands.Program.ApplyMigration(String migration, St
ring context, String startupProject)
   at Microsoft.Data.Entity.Commands.Program.<>c__DisplayClass10_3.<Main>b__9()
   at Microsoft.Framework.Runtime.Common.CommandLine.CommandLineApplication.Exec
ute(String[] args)
   at Microsoft.Data.Entity.Commands.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute(Assembly ass
embly, String[] args, IServiceProvider serviceProvider)
   at Microsoft.Framework.ApplicationHost.Program.ExecuteMain(DefaultHost host,
String applicationName, String[] args)
   at Microsoft.Framework.ApplicationHost.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute(Assembly ass
embly, String[] args, IServiceProvider serviceProvider)
   at dnx.host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, Frame
workName targetFramework)
   at dnx.host.RuntimeBootstrapper.ExecuteAsync(String[] args, FrameworkName tar
getFramework)
   at dnx.host.RuntimeBootstrapper.Execute(String[] args, FrameworkName targetFr
amework)
ClientConnectionId:a6eb058e-c119-4aca-91b3-6518391424b7
Error Number:156,State:1,Class:15

**Comment by maumar** [comment]
will investigate

**Comment by tdekoekkoek** [comment]
Yeah I'm getting this on the latest bits just installed

**Comment by bricelam** [comment]
I haven't investigated, but this should have been fixed (in Beta 8) by #2859 & #2862.

**Comment by devcube** [comment]
I'm getting this same error, is there any known workaround for beta6?

**Comment by natemcmaster** [comment]
@devcube workaround = manually edit the generated migrations files.

**Comment by khellang** [comment]
I'm getting the same on beta7. What was the fix again?

**Comment by bricelam** [comment]
Remove the extraneous `AlterColumn` calls; remove the chained calls to `Annotation("SqlServer:ValueGenerationStrategy", SqlServerIdentityStrategy.IdentityColumn)`

**Comment by khellang** [comment]
Thanks, @bricelam :grin::+1:

**Comment by divega** [comment]
@maumar @rowanmiller @bricelam I must be missing something, but it seems that this is open and in beta 6? :smile: Can you please help me figure out what to do with it?

**Comment by rowanmiller** [comment]
Sounds like it was re-opened on Beta 7 but then should be fixed in Beta 8 (but that hasn't been verified). Clearing the milestone so that we don't lose track of it.

**Comment by rowanmiller** [comment]
It is fixed in #2859 & #2862.
